File: frontend/src/microservices/flight/objects/SearchSuggestionPlannerResponse.ts
// 本文件定义 flight 模块的 `SearchSuggestionPlannerResponse`，作为单条搜索建议响应并提供 JSON 编解码。

export type SearchSuggestionPlannerResponse = {
  resourceType: string
  value: string
  title: string
  subtitle: string
}

export const searchSuggestionPlannerResponseFromJson = (json: string): SearchSuggestionPlannerResponse =>
  JSON.parse(json) as SearchSuggestionPlannerResponse

export const searchSuggestionPlannerResponseToJson = (value: SearchSuggestionPlannerResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/flight/objects/SearchSuggestionPlannerResponse.scala
// SearchSuggestionPlannerResponse defines one flight suggestion row for the search planner.
package com.typesafe.travel.flight.objects

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class SearchSuggestionPlannerResponse(
    resourceType: String,
    value: String,
    title: String,
    subtitle: String
)

object SearchSuggestionPlannerResponse:
  given sourceEncoder: Encoder[SearchSuggestionPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[SearchSuggestionPlannerResponse] = deriveDecoder

